### ISSUE TITLE
Server throwing errrors periodically

### ISSUE THREAD

**Issue by truckerboy** [dahlb/ha_carrier#139]
### System Health details

This error originated from a custom integration.

Logger: custom_components.ha_carrier
Source: helpers/update_coordinator.py:344
integration: Carrier Infinity (documentation, issues)
First occurred: August 15, 2024 at 8:21:29 PM (10 occurrences)
Last logged: 11:31:10 AM

Error fetching ha_carrier-Infinity data: 500 Server Error: Internal Server Error for url: https://www.app-api.ing.carrier.com/users/arnavghosh/activateSystems
Error fetching ha_carrier-Infinity data: 504 Server Error: Gateway Time-out for url: https://www.app-api.ing.carrier.com/users/arnavghosh/activateSystems
Error fetching ha_carrier-Infinity data: 502 Server Error: Bad Gateway for url: https://www.app-api.ing.carrier.com/users/arnavghosh/activateSystems

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of currently [previous issues](https://github.com/dahlb/ha_carrier/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

This error originated from a custom integration.

Logger: custom_components.ha_carrier
Source: helpers/update_coordinator.py:344
integration: Carrier Infinity (documentation, issues)
First occurred: August 15, 2024 at 8:21:29 PM (10 occurrences)
Last logged: 11:31:10 AM

Error fetching ha_carrier-Infinity data: 500 Server Error: Internal Server Error for url: https://www.app-api.ing.carrier.com/users/arnavghosh/activateSystems
Error fetching ha_carrier-Infinity data: 504 Server Error: Gateway Time-out for url: https://www.app-api.ing.carrier.com/users/arnavghosh/activateSystems
Error fetching ha_carrier-Infinity data: 502 Server Error: Bad Gateway for url: https://www.app-api.ing.carrier.com/users/arnavghosh/activateSystems

### Reproduction steps

1.
2.This error originated from a custom integration.

Logger: custom_components.ha_carrier
Source: helpers/update_coordinator.py:344
integration: Carrier Infinity (documentation, issues)
First occurred: August 15, 2024 at 8:21:29 PM (10 occurrences)
Last logged: 11:31:10 AM

Error fetching ha_carrier-Infinity data: 500 Server Error: Internal Server Error for url: https://www.app-api.ing.carrier.com/users/arnavghosh/activateSystems
Error fetching ha_carrier-Infinity data: 504 Server Error: Gateway Time-out for url: https://www.app-api.ing.carrier.com/users/arnavghosh/activateSystems
Error fetching ha_carrier-Infinity data: 502 Server Error: Bad Gateway for url: https://www.app-api.ing.carrier.com/users/arnavghosh/activateSystems
3.
...

### Debug logs

```text
This error originated from a custom integration.

Logger: custom_components.ha_carrier
Source: helpers/update_coordinator.py:344
integration: Carrier Infinity (documentation, issues)
First occurred: August 15, 2024 at 8:21:29 PM (10 occurrences)
Last logged: 11:31:10 AM

Error fetching ha_carrier-Infinity data: 500 Server Error: Internal Server Error for url: https://www.app-api.ing.carrier.com/users/arnavghosh/activateSystems
Error fetching ha_carrier-Infinity data: 504 Server Error: Gateway Time-out for url: https://www.app-api.ing.carrier.com/users/arnavghosh/activateSystems
Error fetching ha_carrier-Infinity data: 502 Server Error: Bad Gateway for url: https://www.app-api.ing.carrier.com/users/arnavghosh/activateSystems
```

**Comment by deviantintegral** [comment]
Yeah, I'm seeing the same sort of thing. It's happening a few times a day and resolves itself. I don't think there's anything for the integration to do, other than perhaps allow users to specify an amount of time to allow for errors to be suppressed.

```
Error fetching ha_carrier-REDACTED data: 503 Server Error: Service Unavailable for url: https://www.app-api.ing.carrier.com/users/REDACTED/activateSystems
Error fetching ha_carrier-Red Cottage data: 502 Server Error: Bad Gateway for url: https://www.app-api.ing.carrier.com/users/REDACTED/activateSystems
Error fetching ha_carrier-Red Cottage data: 504 Server Error: Gateway Timeout for url: https://www.app-api.ing.carrier.com/users/REDACTED/activateSystems
Error fetching ha_carrier-REDACTED data: 502 Server Error: Bad Gateway for url: https://www.app-api.ing.carrier.com/users/REDACTED/activateSystems
Error fetching ha_carrier-REDACTED data: 504 Server Error: Gateway Timeout for url: https://www.app-api.ing.carrier.com/users/REDACTED/activateSystems
```

**Comment by dfludwig** [comment]
I've been disconnected for almost 24 hours now - is that what others are seeing?  I've reset the wifi on my wall thermostat and tried the other tricks I know.  My Carrier Home app is also showing disconnected, so I don't think it's a HA error.

**Comment by deviantintegral** [comment]
I’ve been connected for at least a day now without issue. But, I’m using the Bryant branded version of things.

**Comment by dfludwig** [comment]
I changed the "Polling Interval in Minutes" value in the integration's Configuration from 1 minute to 10 minutes, and have not had connection errors since.  Perhaps coincidence, though....

**Comment by dahlb** [comment]
this seemed to be either a load issue at carrier or poorly implemented rate limiting (I think it was just maintenance issues) either way there isn't really anything for this integration to do about it, but thank you for documenting it here

**Comment by truckerboy** [comment]
> I changed the "Polling Interval in Minutes" value in the integration's Configuration from 1 minute to 10 minutes, and have not had connection errors since. Perhaps coincidence, though....

Could you help pointing out exactly where you found this option?

**Comment by truckerboy** [comment]
> > Could you help pointing out exactly where you found this option?
> 
> Well... there _used_ to be an option/setting for this, but I can't find it now.... :( Perhaps it was depreciated?

These errors keep flooding HA logs. While i dont want them suppressed, we definitely need an explicit option which can help avoid these errors.
